airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -592,20 +604,22 @@ public InputStream getErrorStream() {
   }
 
   /**
-   * Immediately terminates the Kube Pod backing this process and cleans up IO resources.
+   * Waits for the Kube Pod backing this process and returns the exit value after closing resources.
    */
   @Override
   public int waitFor() throws InterruptedException {
-    final Pod refreshedPod =
-        fabricClient.pods().inNamespace(podDefinition.getMetadata().getNamespace()).withName(podDefinition.getMetadata().getName()).get();
-    fabricClient.resource(refreshedPod).waitUntilCondition(KubePodProcess::isTerminal, 10, TimeUnit.DAYS);
-    wasKilled.set(true);
+    try {
+      exitCodeFuture.get();
+    } catch (ExecutionException e) {
+      throw new RuntimeException(e);
```

Comment:
Is this RTE expected? Is it unrecoverable? What does this mean for the waitFor?

Reply:
It means it will catastrophically fail, so the thing calling `waitFor` will need to handle the RTE or fail overall (which is the desired behavior).